src/Symfony/Component/Yaml/Inline.php

```diff
@@ -634,9 +634,18 @@ private static function evaluateScalar(string $scalar, int $flags, array $refere
                         return '0' == $scalar[1] ? -octdec(substr($scalar, 1)) : (($raw === (string) $cast) ? $cast : $raw);
                     case is_numeric($scalar):
                     case Parser::preg_match(self::getHexRegex(), $scalar):
+                    case Parser::preg_match(self::getBinRegex(), $scalar):
                         $scalar = str_replace('_', '', $scalar);
 
-                        return '0x' === $scalar[0].$scalar[1] ? hexdec($scalar) : (float) $scalar;
+                        if ('0x' === $scalar[0].$scalar[1]) {
+                            return hexdec($scalar);
+                        }
+
+                        if ('0b' === $scalar[0].$scalar[1]) {
+                            return bindec($scalar);
+                        }
+
+                        return (float) $scalar;
```

Comment:
I split this return, because there was the case of the binary. Rather than nesting a ternary, I thought it would be better to have 2 proper conditions.